bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestDataClientVeneerApi.java

```diff
@@ -194,6 +199,28 @@ public void testReadRowAsync() throws Exception {
         .futureCall(Mockito.eq(expectedRequest), Mockito.any(GrpcCallContext.class));
   }
 
+  @Test
+  public void testReadRows_Errors() throws IOException {
+    Query query = Query.create(TABLE_ID).rowKey(ROW_KEY);
+    when(mockDataClient.readRowsCallable(Mockito.<RowResultAdapter>any()))
+        .thenThrow(new RuntimeException())
```

Comment:
this doesnt make sense. readRowsCallable() will never throw an exception. You want to test what happens when the server returns an error. I think something like this:

```java
  public void testReadRows_Errors() throws IOException {
    Query query = Query.create(TABLE_ID).rowKey(ROW_KEY);
    when(mockDataClient.readRowsCallable(any(RowResultAdapter.class)))
        .thenReturn(mockStreamingCallable);
    when(mockStreamingCallable.call(any(Query.class), any(GrpcCallContext.class)))
        .thenReturn(serverStream);
    when(serverStream.iterator())
        .thenReturn(new Iterator<Result>() {
          @Override
          public boolean hasNext() {
            return true;
          }
          @Override
          public Result next() {
            throw new InternalException("fake error", null, GrpcStatusCode.of(Code.INTERNAL), false);
          }
        })
        .thenReturn(ImmutableList.<Result>of().iterator());

    assertThrows(Exception.class, () -> dataClientWrapper.readRows(query).next());

    ResultScanner noRowsResultScanner = dataClientWrapper.readRows(query);
    assertNull(noRowsResultScanner.next());
    noRowsResultScanner.close();

    verify(mockDataClient, times(2)).readRowsCallable(Mockito.<RowResultAdapter>any());
    verify(serverStream, times(2)).iterator();
    verify(mockStreamingCallable, times(2))
        .call(any(Query.class), any(GrpcCallContext.class));
  }
```

Reply:
Done